thirdparty/google-styleguide/run_linter.sh

```diff
@@ -14,16 +15,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# ./run_linter <includedir> <srcdir>
-#!/bin/bash
-if [ "$(uname)" == "Darwin" ]; then
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# ./run_linter <includedir1> <includedir2> ... <includedirN> -- <srcdir1> <srcdir2> ... <srcdirN>
```

Comment:
I like this. I was hoping to make these linter changes following our upcoming release. I also like what you did with tests and was toying with that but didn't want to jump to that because I had a full plate with other features and cmake additions I really appreciate this PR as it helps us tremendously. I'll run through it. May not make it through to this release but we'll get this reviewed and merged as it's super helpful.